Index: rEFIt_UEFI/Platform/smbios.c
===================================================================
--- rEFIt_UEFI/Platform/smbios.c	(revision 4849)
+++ rEFIt_UEFI/Platform/smbios.c	(working copy)
@@ -1766,6 +1766,22 @@
 		}
 		gBootStatus = SmbiosTable.Type32->BootStatus;
 }
+
+VOID PatchTableType32 ()
+{
+  SmbiosTable = GetSmbiosTableFromType (EntryPoint, EFI_SMBIOS_TYPE_SYSTEM_BOOT_INFORMATION, 0);
+  if (SmbiosTable.Raw == NULL) {
+    return;
+  }
+  TableSize = SmbiosTableLength(SmbiosTable);
+  ZeroMem((VOID*)newSmbiosTable.Type32, MAX_TABLE_SIZE);
+  CopyMem((VOID*)newSmbiosTable.Type32, (VOID*)SmbiosTable.Type32, TableSize);
+  
+  // Patch System Boot Status
+  newSmbiosTable.Type32->BootStatus = 0;
+  LogSmbiosTable(newSmbiosTable);
+  return;
+}
 						   
 /**
  * Apple Specific Structures.
@@ -2017,6 +2033,7 @@
    PatchTableType16();
 	PatchTableType19();
 	PatchTableType20();
+  PatchTableType32();
 	PatchTableType128();
 	PatchTableType130();
 	PatchTableType131();
