Index: rEFIt_UEFI/Platform/smbios.c
===================================================================
--- rEFIt_UEFI/Platform/smbios.c	(revision 4849)
+++ rEFIt_UEFI/Platform/smbios.c	(working copy)
@@ -764,8 +764,9 @@
 		CopyMem((VOID*)newSmbiosTable.Type4, (VOID*)SmbiosTable.Type4, Size); //copy main table
 		CopyMem((CHAR8*)newSmbiosTable.Type4+NewSize, (CHAR8*)SmbiosTable.Type4+Size, TableSize - Size); //copy strings
 		newSmbiosTable.Type4->Hdr.Length = (UINT8)NewSize;
-
-		newSmbiosTable.Type4->MaxSpeed = (UINT16)gCPUStructure.MaxSpeed;	
+    
+    newSmbiosTable.Type4->MaxSpeed = (UINT16)(DivU64x32(gCPUStructure.ExternalClock * MaxRatio, kilo) / 10);
+    newSmbiosTable.Type4->CurrentSpeed = newSmbiosTable.Type4->MaxSpeed;
 		//old version has no such fields. Fill now
 		if (Size <= 0x20){
 			//sanity check and clear
