diff --git a/rEFIt_UEFI/Platform/Settings.h b/rEFIt_UEFI/Platform/Settings.h
index 98c89623c..4f9513a78 100644
--- a/rEFIt_UEFI/Platform/Settings.h
+++ b/rEFIt_UEFI/Platform/Settings.h
@@ -873,6 +873,9 @@ GetCrc32 (
   UINTN Size
   );
 
+VOID
+CorrectCPUProperties (VOID);
+
 VOID
 GetDevices(VOID);
 
diff --git a/rEFIt_UEFI/Platform/cpu.cpp b/rEFIt_UEFI/Platform/cpu.cpp
index bb30c4d7d..59b57f896 100644
--- a/rEFIt_UEFI/Platform/cpu.cpp
+++ b/rEFIt_UEFI/Platform/cpu.cpp
@@ -1162,6 +1162,29 @@ VOID GetCPUProperties (VOID)
   //  return;
 }
 
+VOID CorrectCPUProperties (VOID)
+{
+  UINT16  FsbMhzArray[8] = { 266, 133, 200, 166, 333, 100, 400, 233 };
+  UINT64  Msr = 0;
+  BOOLEAN Corrected = TRUE;
+
+  switch (gCPUStructure.Model)
+  {
+    case CPU_MODEL_PENRYN: // CPUID: 1067x
+      Msr = AsmReadMsr64(MSR_FSB_FREQ);
+      gCPUStructure.ExternalClock = MultU64x32(FsbMhzArray[Msr & 7], kilo);
+      break;
+    default:
+      Corrected = FALSE;
+      break;
+  }
+
+  if (Corrected)
+  {
+    DBG("Corrected ExternalClock: %d MHz\n", (INT32)(DivU64x32(gCPUStructure.ExternalClock, kilo)));
+  }
+}
+
 VOID SetCPUProperties (VOID)
 {
   UINT64    msr = 0;
diff --git a/rEFIt_UEFI/refit/main.cpp b/rEFIt_UEFI/refit/main.cpp
index ca237b81d..7f10498bc 100644
--- a/rEFIt_UEFI/refit/main.cpp
+++ b/rEFIt_UEFI/refit/main.cpp
@@ -2174,6 +2174,8 @@ RefitMain (IN EFI_HANDLE           ImageHandle,
       break;
   }
 
+  CorrectCPUProperties();
+    
   if (!GlobalConfig.NoEarlyProgress && !GlobalConfig.FastBoot && GlobalConfig.Timeout>0) {
     XStringW Message = SWPrintf("... user settings ...");
     BootScreen.EraseTextXY();
