Index: rEFIt_UEFI/Platform/DataHubCpu.c
===================================================================
--- rEFIt_UEFI/Platform/DataHubCpu.c	(revision 4849)
+++ rEFIt_UEFI/Platform/DataHubCpu.c	(working copy)
@@ -427,8 +427,8 @@
     }
 
     TscFrequency        = 0; //gCPUStructure.TSCFrequency;
-    LogDataHub(&gEfiProcessorSubClassGuid, L"InitialTSC",       &TscFrequency,        sizeof(UINT64));
-    LogDataHub(&gEfiProcessorSubClassGuid, L"CPUFrequency",     &CpuSpeed,            sizeof(UINT64));
+    //LogDataHub(&gEfiProcessorSubClassGuid, L"InitialTSC",       &TscFrequency,        sizeof(UINT64));
+    //LogDataHub(&gEfiProcessorSubClassGuid, L"CPUFrequency",     &CpuSpeed,            sizeof(UINT64));
 
     //gSettings.BoardNumber
     LogDataHub(&gEfiMiscSubClassGuid,      L"board-id",         &gSettings.BoardNumber,   (UINT32)iStrLen(gSettings.BoardNumber, 64) + 1);
