diff --git a/rEFIt_UEFI/Platform/DataHubCpu.cpp b/rEFIt_UEFI/Platform/DataHubCpu.cpp
index 3b11cb794..217ddb1cc 100644
--- a/rEFIt_UEFI/Platform/DataHubCpu.cpp
+++ b/rEFIt_UEFI/Platform/DataHubCpu.cpp
@@ -495,8 +495,6 @@ SetupDataForOSX(BOOLEAN Hibernate)
     }
 
     TscFrequency        = 0; //gCPUStructure.TSCFrequency;
-    LogDataHub(&gEfiProcessorSubClassGuid, L"InitialTSC",       &TscFrequency,        sizeof(UINT64));
-    LogDataHub(&gEfiProcessorSubClassGuid, L"CPUFrequency",     &CpuSpeed,            sizeof(UINT64));
 
     //gSettings.BoardNumber
     LogDataHubXString8(&gEfiMiscSubClassGuid,      L"board-id",         gSettings.BoardNumber);
